pkg/controller/service/servicecontroller.go

```diff
@@ -741,6 +741,7 @@ func (s *ServiceController) lockedUpdateLoadBalancerHosts(service *api.Service,
 		return nil
 	}
 
+	hosts := hostsFromNodeList(nodes)
```

Comment:
Do we need to record the hosts here... Elsewhere the nodeList items are used.
If this was removed here and use of of it in createLoadBalancer() above was refatored, then then hostsFromNodeList() could be remove the service package leaving only the HostsFromNodeList (one imeplementation) in the cloud provider api package
(A tests for hostsFromNodeList() could also be removed from servicecontroller_test.go)